TestKMSv2Healthz flake

### Which jobs are flaking?

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/117614/pull-kubernetes-integration/1651530152728858624



### Which tests are flaking?

{Failed  === RUN   TestKMSv2Healthz
--- FAIL: TestKMSv2Healthz (167.38s)


also shown in https://storage.googleapis.com/k8s-metrics/flakes-latest.json
      "k8s.io/kubernetes/test/integration/controlplane/transformation TestKMSv2Healthz": 1,


### Since when has it been flaking?

Not sure

### Testgrid link

_No response_

### Reason for failure (if possible)

```
E0427 10:46:16.817671   85838 controller.go:259] unable to sync kubernetes service: Endpoints "kubernetes" is invalid: subsets[0].addresses[0].ip: Invalid value: "127.0.0.1": may not be in the loopback range (127.0.0.0/8, ::1/128)
I0427 10:46:16.911127   85838 transformation_test.go:108] Started kube-apiserver https://127.0.0.1:34851
    transformation_test.go:560: expected server check "/kms-providers" to be unhealthy with message "internal server error: kms-provider-0: rpc error: code = FailedPrecondition desc = failed precondition - key disabled" but it is not: ok
    transformation_test.go:560: expected server check "/kms-providers" to be unhealthy with message "internal server error: kms-provider-0: rpc error: code = FailedPrecondition desc = failed precondition - key disabled" but it is not: ok
```


### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig auth

--- a/test/integration/controlplane/transformation/transformation_test.go
+++ b/test/integration/controlplane/transformation/transformation_test.go
@@ -528,7 +528,7 @@ func (e *transformTest) printMetrics() error {
 func mustBeHealthy(t kubeapiservertesting.Logger, checkName, wantBodyContains string, clientConfig *rest.Config, excludes ...string) {
 	t.Helper()
 	var restErr error
-	pollErr := wait.PollImmediate(2*time.Second, 2*time.Minute, func() (bool, error) {
+	pollErr := wait.PollImmediate(1*time.Second, 5*time.Minute, func() (bool, error) {
 		body, ok, err := getHealthz(checkName, clientConfig, excludes...)
 		restErr = err
 		if err != nil {
@@ -541,7 +541,7 @@ func mustBeHealthy(t kubeapiservertesting.Logger, checkName, wantBodyContains st
 		return done, nil
 	})
 
-	if pollErr == wait.ErrWaitTimeout {
+	if pollErr != nil {
 		t.Fatalf("failed to get the expected healthz status of OK for check: %s, error: %v, debug inner error: %v", checkName, pollErr, restErr)
 	}
 }
@@ -549,7 +549,7 @@ func mustBeHealthy(t kubeapiservertesting.Logger, checkName, wantBodyContains st
 func mustBeUnHealthy(t kubeapiservertesting.Logger, checkName, wantBodyContains string, clientConfig *rest.Config, excludes ...string) {
 	t.Helper()
 	var restErr error
-	pollErr := wait.PollImmediate(2*time.Second, 2*time.Minute, func() (bool, error) {
+	pollErr := wait.PollImmediate(1*time.Second, 5*time.Minute, func() (bool, error) {
 		body, ok, err := getHealthz(checkName, clientConfig, excludes...)
 		restErr = err
 		if err != nil {
@@ -562,7 +562,7 @@ func mustBeUnHealthy(t kubeapiservertesting.Logger, checkName, wantBodyContains 
 		return done, nil
 	})
 
-	if pollErr == wait.ErrWaitTimeout {
+	if pollErr != nil {
 		t.Fatalf("failed to get the expected healthz status of !OK for check: %s, error: %v, debug inner error: %v", checkName, pollErr, restErr)
 	}
 }

